libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -852,9 +859,13 @@ void AP_DDS_Client::on_request(uxrSession* uxr_session, uxrObjectId object_id, u
         if (deserialize_success == false) {
             break;
         }
-        mode_switch_response.status = AP::vehicle()->set_mode(mode_switch_request.mode, ModeReason::DDS_COMMAND);
         mode_switch_response.curr_mode = AP::vehicle()->get_mode();
-
+        if (external_control->is_enabled()) {
```

Comment:
This logic can get cleaned up once https://github.com/ArduPilot/ardupilot/pull/28696 goes in.